src/npc.cpp

```diff
@@ -294,6 +302,19 @@ void Npc::onCreatureMove(Creature* creature, const Tile* newTile, const Position
 		if (m_npcEventHandler) {
 			m_npcEventHandler->onCreatureMove(creature, oldPos, newPos);
 		}
+
+		if (creature != this) {
+			Player* player = creature->getPlayer();
+
+			// if player is now in range, add to spectators list, otherwise erase
+			if (player->canSee(_position)) {
```

Comment:
Seriously, guys? The diff only shows part of the code. The line just above the diff is:
![Imgur](http://i.imgur.com/EhwMYuy.png)

So, if `creature != this`, then creature IS player.